gsy_framework/sim_results/carbon_emissions.py

```diff
@@ -0,0 +1,169 @@
+                .set_index("newlevel", append=True)
+                .unstack("newlevel")
+            )
+        df = df.truncate(before=start, after=end)
+        return df
+
+    def _calculate_total_carbon_generated(
+        self, country_code: str, start: pd.Timestamp, end: pd.Timestamp, stat: str = "median"
+    ) -> pd.DataFrame:
+        """Calculate the total carbon generated based on the specified statistic
+        (min, median, max)"""
+        if stat not in ["min", "median", "max"]:
+            raise ValueError("stat must be one of 'min', 'median', or 'max'")
+
+        df = self._query_generation_per_plant(country_code, start, end)
+
+        generation_types = df.iloc[0].iloc[1:]
+        carbon_emission_per_plant = lambda x: GENERATION_PLANT_TO_CARBON_EMISSIONS[x][
+            stat
+        ]  # noqa: E731 C3001
+        emissions_map = generation_types.map(carbon_emission_per_plant)
+
+        df_numeric = df.iloc[2:].reset_index(drop=True)
+        df_numeric = df_numeric.set_index("Unnamed: 0").apply(  # noqa: E501
+            pd.to_numeric, errors="coerce"
+        )
+        df_numeric.index = pd.to_datetime(df_numeric.index)
+        df_numeric["Total Energy (kWh)"] = df_numeric.sum(axis=1) * 1000  # convert MWh to kWh
+        df_numeric["Total Carbon (gCO2eq)"] = df_numeric.mul(emissions_map, axis=1).sum(axis=1)
+
+        generation_types = df.iloc[0]
+        emissions_map = generation_types.map(
+            lambda x: GENERATION_PLANT_TO_CARBON_EMISSIONS.get(x, {}).get(stat, 0)
+        )
+        return df_numeric[["Total Energy (kWh)", "Total Carbon (gCO2eq)"]]
+
+    def update(self, country_code: str, time_slot: duration, results_list: List[Dict]):
+        """Update the results_dict with carbon generated and carbon savings"""
+
+        start_date = pd.Timestamp(results_list[0]["current_market"].split("T")[0])
+        end_date = pd.Timestamp(results_list[-1]["current_market"].split("T")[0])
+        df_total_carbon_and_energy = self._calculate_total_carbon_generated(
+            country_code, start_date, end_date
+        )
+        for result in results_list:
+            current_market = result["current_market"]
+
+            for area_uuid in result["results"].keys():
+                area_result = result["results"][area_uuid]
+                target_timestamp = pd.Timestamp(current_market).tz_localize("UTC")
+
+                nearest_index = df_total_carbon_and_energy.index.get_indexer(
+                    [target_timestamp], method="nearest"
+                )[0]
+                nearest_row = df_total_carbon_and_energy.iloc[nearest_index]
+
+                total_carbon_g = nearest_row["Total Carbon (gCO2eq)"]
+                total_energy_kWh = nearest_row["Total Energy (kWh)"]
+
+                imported_energy_from_grid_kWh = area_result["bought_from_grid"]
+                imported_energy_from_community_kWh = area_result[
+                    "energy_bought_from_community_kWh"
```

Comment:
With this, the carbon foodprint is only applicable for SCM. Don't we want to use also the `ImportedExportedEnergyHandler` or instead?

Reply:
Yes, how do I adapt to use what you are saying. Let's catch up after standup.

Reply:
This is indeed not needed BTW if you use the ImportedExportedEnergyHandler

Reply:
I will need this method in gsy-e CLI